app/models/user.rb

```diff
@@ -1,10 +1,30 @@
 class User < ApplicationRecord
-  has_many :tasks
+  before_destroy :destroy_action
+  before_update :update_action
+  has_many :tasks, dependent: :destroy # "dependent: :destroy" はレコード削除時に関連すtasksも削除する(callbackあり)
   validates :name, presence: true, length: { maximum: 30 }
   validates :email, presence: true, length: { maximum: 255 },
                     format: { with: /\A[\w+\-.]+@[a-z\d\-.]+\.[a-z]+\z/i }
   
   before_validation { email.downcase! }
   has_secure_password
   validates :password, presence: true, length: { minimum: 6 }
+
+  private
+
+  def destroy_action
+    # 最後の1管理者の場合は削除できない。(admin:trueの数が1以下になってはいけない, 現状自分の最後の1人ということは今自分のadminがtrueであるはず)
+    if User.where(admin: true).count <= 1 && self.admin == true
+      throw :abort
+    end
+  end
+
+  def update_action
+    # 最後の1管理者の場合は管理者をfalseに変更できない。(admin:trueの数が1以下になってはいけないので、admin trueが1人の時のupdateの時に、admin falseにできてないけない)
+    if User.where(admin: true).count == 1 && self.admin == true
```

Comment:
下記のように記載できます。またこの判定メソッドは他でも使っているのでメソッドに切り出して呼び出す形にしましょう。
```suggestion
    if User.admin.one? && self.admin?
```
[参考] https://docs.ruby-lang.org/ja/latest/method/Enumerable/i/one=3f.html